### PR TITLE
RHDEVDOCS-3678 - Add section for direct access to 3rd party monitoring components' service APIs

### PR DESCRIPTION
**Pull request by bburt-rh**
This PR does the following:

- Reorganizes, renames, and rewrites the previous version section, "Accessing third-party UIs" to "Accessing third-party UIs and APIs"
- documents direct CLI access to 3rd party monitoring components' service APIs 
- updates earlier changes relating to the removal of the Prometheus web UI and adds a deprecation notice for removal of the Prometheus and Grafana web UIs in the next release

Details:
- Aligned team: Dev Tools
- For branches: 4.10+
- Jira: https://issues.redhat.com/browse/RHDEVDOCS-3678, https://issues.redhat.com/browse/RHDEVDOCS-3424
- Direct link to doc preview: https://deploy-preview-41384--osdocs.netlify.app/openshift-enterprise/latest/monitoring/accessing-third-party-monitoring-uis-and-apis.html
- SME review: @raptorsun (approved) @simonpasquier (approved)
- QE review: @juzhao (approved) @lihongyan1 (approved)
- Peer review: @rolfedh (approved)
- <All reviews complete. Please merge now>

### CODE REVIEW

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (bburt-rh) [inline modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:13]: @raptorsun Can you suggest some wording here for adding the information you gave me about the minimum permissions needed in Oauth-proxy for Alertmanager?

Comment (raptorsun) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:13]: The access setting comes from [this argument](https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/alertmanager/alertmanager.yaml#L32). We may explain the setting as below:

`Openshift uses Subject Access Review for Alertmanager. It requires the user to have at least "patch" permission to resource "alertmanager" in namespace "openshift-monitoring".`

Could you please double check? @jan--f  I'm not so familiar with [Subject Access Review](https://kubernetes.io/docs/reference/access-authn-authz/authorization/) and may miss some details 😅

File: _topic_maps/_topic_map.yml
Comment (simonpasquier) [inline _topic_maps/_topic_map.yml:2016]:
```suggestion
- Name: Accessing third-party monitoring UIs and APIs
```

File: modules/monitoring-accessing-third-party-monitoring-uis.adoc
Comment (simonpasquier) [inline modules/monitoring-accessing-third-party-monitoring-uis.adoc:10]: we missed the deadline for Prometheus so it's going to be possible accessing its UI in 4.10.

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (simonpasquier) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:13]: It might be more convenient to say that the associated account needs to be bound against the `monitoring-alertmanager-edit` role in the `openshift-monitoring` namespace (this way the exact permissions may change in the future without having to reword the documentation).

Comment (raptorsun) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:13]: Agree. We just tell users to bond to this role instead of listing permissions. Easier to understand for users.

File: _topic_maps/_topic_map.yml
Comment (bburt-rh) [reply _topic_maps/_topic_map.yml:2016]: fixed

File: modules/monitoring-accessing-third-party-monitoring-uis.adoc
Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-uis.adoc:10]: fixed

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:13]: @simonpasquier @raptorsun Is this same account binding also required for the components other than Alertmanager? Or do these other components require that the account be bound to a different role?

**Comment by juzhao** [comment]
@lihongyan1 please help to review

File: monitoring/accessing-third-party-monitoring-uis-and-apis.adoc
Comment (juzhao) [inline monitoring/accessing-third-party-monitoring-uis-and-apis.adoc:10]: we still can access Grafana UI in 4.10

Comment (juzhao) [inline monitoring/accessing-third-party-monitoring-uis-and-apis.adoc:10]: we still can access Grafana UI in 4.10

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (juzhao) [inline modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:10]: no need to mention Prometheus here, since Prometheus UI is still could be accessed in 4.10

File: modules/monitoring-accessing-third-party-monitoring-uis.adoc
Comment (lihongyan1) [inline modules/monitoring-accessing-third-party-monitoring-uis.adoc:10]: since 'Thanos Ruler' is in  [accessing-third-party-monitoring-uis-and-apis.adoc](https://github.com/openshift/openshift-docs/pull/41384/files#diff-2d645834c95c663c4aecf714e32dc362a918117b35f8866e55508e2e80a34709), it should be added here also.  Grafana UI is still accessible till OCP 4.10. 
does not provide or support direct access to third-party web user interfaces (UIs) for components in the monitoring stack such as Alertmanager, Thanos Ruler, or Thanos Querier.

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (lihongyan1) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:10]: > no need to mention Prometheus here, since Prometheus UI is still could be accessed in 4.10

Prometheus API is also accessible.

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (juzhao) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:13]: 1. I am afraid the `monitoring-alertmanager-edit` role is not enough to see the alertmanager APIs, the role is not able to execute `host=$(oc -n openshift-monitoring get route alertmanager-main -ojsonpath={.spec.host})`, if the user know the alertmanager route, then it is not needed to get the route, there is not such issue
2. The `monitoring-alertmanager-edit` role is not able to view thanos-querier service APIs
```
# oc adm policy add-role-to-user monitoring-alertmanager-edit testuser-11 -n openshift-monitoring --role-namespace openshift-monitoring
role.rbac.authorization.k8s.io/monitoring-alertmanager-edit added: "testuser-11"

# oc login https://api.***.com:6443 -u testuser-11 -p ***
The server uses a certificate signed by an unknown authority.
You can bypass the certificate check, but any data you send to the server could be intercepted by others.
Use insecure connections? (y/n): y

Login successful.

You don't have any projects. You can try to create a new project, by running

    oc new-project <projectname>

# host=$(oc -n openshift-monitoring get route alertmanager-main -ojsonpath={.spec.host})
Error from server (Forbidden): routes.route.openshift.io "alertmanager-main" is forbidden: User "testuser-11" cannot get resource "routes" in API group "route.openshift.io" in the namespace "openshift-monitoring"

### if we already know the alertmanager route, no such issue:
# host=alertmanager-main-openshift-monitoring.*****
# token=#(oc whoami -t)
# curl -H "Authorization: Bearer #token" -k "https://#host/api/v2/receivers"
[{"name":"Default"},{"name":"Watchdog"},{"name":"Critical"}]

```

Comment (raptorsun) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:13]: - Alertmanager: the role `monitoring-alertmanager-edit` grants access to alert manager API
- Thanos Ruler & Querier: The requesting account must have "get" permission on the namespace that it is querying. The user may create their roles and grant them access to certain namespaces. For us we cannot give an exhaust list of these roles in advance. 
What do you think? @simonpasquier

Comment (juzhao) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:13]: we should change the sequence to
```
The following example shows how to query the service API receivers for Alertmanager

This access requires that the associated user account be bound against the monitoring-alertmanager-edit role in the openshift-monitoring namespace and only supports using a Bearer Token for authentication.
```
and add note, for other service APIs, the requesting account must have "get" permission on the namespace

File: modules/monitoring-accessing-third-party-monitoring-uis.adoc
Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-uis.adoc:10]: Added Thanos ruler

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:10]: I'm keeping the reference to Prometheus because it's API is also accessible.

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:13]: I updated the text as follows:

```
The following example shows how to query the service API receivers for Alertmanager.
This example requires that the associated user account be bound against the `monitoring-alertmanager-edit` role in the `openshift-monitoring` namespace and only supports using a Bearer Token for authentication.

[source, terminal]
----
$ host=$(oc -n openshift-monitoring get route alertmanager-main -ojsonpath={.spec.host})
$ token=$(oc whoami -t)
$ curl -H "Authorization: Bearer $token" -k "https://$host/api/v2/receivers"
----

To access Thanos Ruler and Thanos Querier service APIs, the requesting account must have `get` permission for the namespace to be queried.
```

File: monitoring/accessing-third-party-monitoring-uis-and-apis.adoc
Comment (bburt-rh) [reply monitoring/accessing-third-party-monitoring-uis-and-apis.adoc:10]: fixed

**Comment by bburt-rh** [comment]
@lihongyan1 @raptorsun @simonpasquier @juzhao - Please have another look at this PR and provide any further feedback. If no further changes are required, please provide an lgtm. Thanks!

**Comment by juzhao** [comment]
LGTM, just `monitoring-alertmanager-edit` role is not enough to see the alertmanager APIs, the role is not able to execute 
`host=$(oc -n openshift-monitoring get route alertmanager-main -ojsonpath={.spec.host})`, tried in https://github.com/openshift/openshift-docs/pull/41384#discussion_r807687996
we should also add it requires the account to view the route

**Comment by bburt-rh** [comment]
> LGTM, just `monitoring-alertmanager-edit` role is not able to execute `host=$(oc -n openshift-monitoring get route alertmanager-main -ojsonpath={.spec.host})`, tried in [#41384 (comment)](https://github.com/openshift/openshift-docs/pull/41384#discussion_r807687996) we should also add it requires the account has the privilege to view the route

Added a phrase to say that it requires that the account also has the privilege to view the route

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (rolfedh) [inline modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:23]: Move this to the prerequisites section?

File: monitoring/accessing-third-party-monitoring-uis-and-apis.adoc
Comment (rolfedh) [inline monitoring/accessing-third-party-monitoring-uis-and-apis.adoc:20]: If these are additional resources for the assembly, not the preceding module, apply the following formatting instead.
```suggestion
== Additional resources
```
See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#additional-resources-sections

File: monitoring/monitoring-overview.adoc
Comment (rolfedh) [inline monitoring/monitoring-overview.adoc:21]: Does it make sense to use this continuously updating attribute, `{product-version}`, here? 
Features described by each release of the documentation are release-specific. Maybe we can remove this attribute?

File: monitoring/monitoring-overview.adoc
Comment (rolfedh) [inline monitoring/monitoring-overview.adoc:22]: Should the section titles in all of these xrefs use sentence case, for example, `[troubleshooting monitoring issues]`?

File: monitoring/reviewing-monitoring-dashboards.adoc
Comment (rolfedh) [inline monitoring/reviewing-monitoring-dashboards.adoc:49]: It looks like there's a new requirement to anchor these assembly-level sections: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies

File: monitoring/monitoring-overview.adoc
Comment (bburt-rh) [reply monitoring/monitoring-overview.adoc:21]: I removed the version attribute. You're right. It's not needed here.

File: monitoring/monitoring-overview.adoc
Comment (bburt-rh) [reply monitoring/monitoring-overview.adoc:22]: Yes. Fixed.

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:23]: It's really more of an "additional information" sentence to note a difference with accessing these components versus what is being accessed in the example code. I've set it off with a Note admonition because of how it's functioning here.

File: monitoring/accessing-third-party-monitoring-uis-and-apis.adoc
Comment (bburt-rh) [reply monitoring/accessing-third-party-monitoring-uis-and-apis.adoc:20]: Fixed, and added the attributes for Additional Resources sections per the guidance.

File: monitoring/reviewing-monitoring-dashboards.adoc
Comment (bburt-rh) [reply monitoring/reviewing-monitoring-dashboards.adoc:49]: Added the anchor attribute per the guidance in the link

**Review by rolfedh — APPROVED** [review]
Please remember to fetch, rebase, and rebuild locally without errors before requesting merge.

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (simonpasquier) [inline modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:25]: We could also mention the `cluster-monitoring-view` clusterrole.
```suggestion
To access Thanos Ruler and Thanos Querier service APIs, the requesting account must have the `get` permission on the namespaces resource.
```

**Review by raptorsun — APPROVED** [review]
/lgtm

File: modules/monitoring-accessing-third-party-monitoring-uis.adoc
Comment (simonpasquier) [inline modules/monitoring-accessing-third-party-monitoring-uis.adoc:15]: can we say that it's going to be removed in the next release (instead of "a future release)?

Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-uis.adoc:15]: > can we say that it's going to be removed in the next release (instead of "a future release)?

@simonpasquier Are we 100% certain it will be removed in the next release? In general, we try to avoid referring to specific future releases, but in this case it might be ok.

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:25]: > We could also mention the `cluster-monitoring-view` clusterrole.

@simonpasquier Would this wording work? "To access Thanos Ruler and Thanos Querier service APIs, the requesting account must have the `get` permission on the namespaces resource and be assigned the `cluster-monitoring-view` cluster role."

File: modules/monitoring-accessing-third-party-monitoring-uis.adoc
Comment (simonpasquier) [reply modules/monitoring-accessing-third-party-monitoring-uis.adoc:15]: Pretty sure it will be. The feedback we had before is that deprecating Grafana without a firm date wasn't nice for users. Maybe express that the UIs could be removed in 4.11 at the earliest?

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (simonpasquier) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:25]: Both parts are synonyms. Maybe: "To access Thanos Ruler and Thanos Querier service APIs, the requesting account must have the `get` permission on the namespaces resource which can be done by granting the `cluster-monitoring-view` cluster role."

File: modules/monitoring-accessing-third-party-monitoring-uis.adoc
Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-uis.adoc:15]: Will change it to "next release"

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc
Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc]: Fixed wording per your suggestion

**Pull request merged**
